pycargoebuild/__main__.py

```diff
@@ -71,13 +71,24 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def locate_workspace_root(directory: Path) -> Path:
+        for current_dir in (directory.resolve() / "Cargo.lock" ).parents:
+            try:
+                with open(current_dir / "Cargo.lock", "rb"):
+                    return current_dir
+            except FileNotFoundError as e:
+                err = e
```

Comment:
I don't think you need to use an extra variable for this (but I could be wrong).

Reply:
It does not work correctly without this variable. The code crashes with
```
 UnboundLocalError: local variable 'e' referenced before assignment
```
exception when I remove `err = e` line and change `raise err` to `raise e`.